### PR TITLE
Improve mining dispatcher

### PR DESCRIPTION
**Pull request by polarker**
* Use mining dispatcher for mining connections actors
* Refactor a bit

### CODE REVIEW

File: flow/src/main/scala/org/alephium/flow/mining/MinerApiController.scala [hunk 1]
```diff
@@ -51,7 +51,7 @@ object MinerApiController {
       groupConfig: GroupConfig,
       networkSetting: NetworkSetting
   ): Props = {
-    Props(new MyConnectionHandler(remote, connection))
+    Props(new MyConnectionHandler(remote, connection)).withDispatcher(MiningDispatcher)
```
Comment (polarker) [inline flow/src/main/scala/org/alephium/flow/mining/MinerApiController.scala:54]: This is the key change

**Review by Lbqds — APPROVED** [review]

File: flow/src/test/scala/org/alephium/flow/mining/MinerSpec.scala [hunk 2]
```diff
@@ -20,14 +20,15 @@ import scala.annotation.tailrec
...
 import org.scalatest.concurrent.ScalaFutures
 
 import org.alephium.flow.{AlephiumFlowActorSpec, FlowFixture}
 import org.alephium.flow.handler.{BlockChainHandler, TestUtils, ViewHandler}
 import org.alephium.flow.model.MiningBlob
 import org.alephium.flow.validation.InvalidBlockVersion
-import org.alephium.protocol.model._
-import org.alephium.serde._
+import org.alephium.protocol.model.*
+import org.alephium.serde.*
```
Comment (tdroxler) [inline flow/src/test/scala/org/alephium/flow/mining/MinerSpec.scala:31]: should we restrict to `_` for consistency?

**Review by tdroxler — APPROVED** [review]

File: flow/src/test/scala/org/alephium/flow/mining/MinerSpec.scala (hunk 2 above)
Comment (polarker) [reply flow/src/test/scala/org/alephium/flow/mining/MinerSpec.scala:31]: yes, I will fix this

Comment (polarker) [reply flow/src/test/scala/org/alephium/flow/mining/MinerSpec.scala]: Fixed!

**Pull request merged**